### PR TITLE
Pv2 2160 inform data collections if metrics tolerance is met

### PR DESCRIPTION
**Pull request by tom-gough**

### CODE REVIEW

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Payments.Monitoring.ServiceFabric/ApplicationPackageRoot/ApplicationManifest.xml [hunk 1]
```diff
@@ -39,6 +39,8 @@
     <Parameter Name="DelayedMessageRetries" DefaultValue="3" />
     <Parameter Name="DelayedMessageRetryDelay" DefaultValue="00:00:10" />
     <Parameter Name="MetricsGenerationDelay" DefaultValue="00:05:00" />
+    <Parameter Name="FunctionApiKey" DefaultValue="" />
+    <Parameter Name="FunctionBaseUrl" DefaultValue="" />
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.Monitoring.ServiceFabric/ApplicationPackageRoot/ApplicationManifest.xml:43]: Please rename these configs to SubmissionWindowValidationFunction* (ApiKey, BaseUrl)

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

**Review by hardik-desai-kcl — DISMISSED** [review]

**Review by hardik-desai-kcl — APPROVED** [review]

**Review by hardik-desai-kcl — DISMISSED** [review]

**Review by carlbm — DISMISSED** [review]

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+            mocker.Mock<IMetricsValidationService>()
+                .Setup(x => x.Validate(jobId,It.IsAny<short>(),It.IsAny<byte>()))
+                .ReturnsAsync(false);
+
+            var service = mocker.Create<PeriodEndSubmissionWindowValidationJobStatusService>();
+            
+            var result = await service.ManageStatus(jobId, CancellationToken.None).ConfigureAwait(false);
+            
+            result.Should().BeTrue();
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs:59]: Do we care about this? It diverts some attention from the important part of the test

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        [Test]
+        public async Task ManageStatus_When_MetricsValidationService_Returns_False_Then_CompletesWithCompletedWithErrorsStatus()
+        {
+            const int jobId = 99;
+
+            mocker.Mock<IMetricsValidationService>()
+                .Setup(x => x.Validate(jobId,It.IsAny<short>(),It.IsAny<byte>()))
+                .ReturnsAsync(false);
+
+            var service = mocker.Create<PeriodEndSubmissionWindowValidationJobStatusService>();
```
Comment (carlbm) [inline src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs:55]: can we call the system under test `sut`? We do in other parts of the solution

**Review by carlbm — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs (hunk 2 above)
Comment (hardik-desai-kcl) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs:59]: we do care about this being true as it's reported back upstream classes

File: src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs (hunk 3 above)
Comment (hardik-desai-kcl) [reply src/SFA.DAS.Payments.Monitoring.Jobs.Application.UnitTests/JobProcessing/PeriodEnd/PeriodEndSubmissionWindowValidationJobStatusServiceTests.cs]: changed

**Review by carlbm — APPROVED** [review]

**Review by carlbm — APPROVED** [review]

**Pull request merged**